tensorboard/webapp/app_routing/route_config.ts

```diff
@@ -249,20 +313,24 @@ export class RouteConfigs {
 
       this.configMatchers.push(matcher);
 
-      if (matcher.config.routeKind !== null) {
-        this.routeKindToConfigMatchers.set(matcher.config.routeKind, matcher);
+      if (matcher.config.routeType === RouteType.CONCRETE) {
+        this.routeKindToConfigMatchers.set(
+          matcher.config.definition.routeKind,
+          matcher
+        );
 
-        if (matcher.config.defaultRoute) {
+        if (matcher.config.definition.defaultRoute) {
```

Comment:
Now that we have `ConcreteRouteConfigMatcher`, would it make sense to drop L419, the part that checks at runtime whether `defaultRouteConfig` is a CONCRETE type, and instead do a check here `isConcreteRoute*`?  Then, we can more strictly type the field `defaultRouteConfig: ConcreteRouteConfigMatcher | null;`